<a href="https://colab.research.google.com/github/Aashika2001/Trial/blob/master/RandomForestAssignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### #IMPORTING THE NECESSARY PACKAGES

In [1]:
from tensorflow import keras
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd

In [2]:
from sklearn.metrics import accuracy_score,confusion_matrix,classification_report
import numpy as np

In [3]:
import cv2


#LOADING THE DATASET




In [4]:
(x_train, y_train), (x_test, y_test) = keras.datasets.cifar10.load_data()

170508288/170498071 [==============================] - 4s 0us/step


### #OBTAINING THE SIZE OF THE TEST AND TRAIN DATA SETS

In [5]:
x_train.shape,x_test.shape

((50000, 32, 32, 3), (10000, 32, 32, 3))

In [6]:
#Preprocessing the data - to maintain common pixel range for all images
x_train=x_train/255.0
x_test=x_test/255.0

In [7]:
nsamples, nx, ny, nrgb = x_train.shape
x_train2 = x_train.reshape((nsamples,nx*ny*nrgb))

In [8]:
nsamples, nx, ny, nrgb = x_test.shape
x_test2 = x_test.reshape((nsamples,nx*ny*nrgb))

In [9]:
x_train2.shape,x_test2.shape

((50000, 3072), (10000, 3072))

### #Implementing a Random Forest Classifier - import it from sklearn

In [10]:
from sklearn.ensemble import RandomForestClassifier

In [11]:
#creating an instance
model=RandomForestClassifier()

In [12]:
#train the model
model.fit(x_train2,y_train)

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  """Entry point for launching an IPython kernel.


RandomForestClassifier()

In [13]:
#predict for the test set using the fitted Random Forest Classifier model
y_pred=model.predict(x_test2)
y_pred

array([3, 8, 8, ..., 5, 5, 7], dtype=uint8)

In [14]:
accuracy_score(y_pred,y_test)
print(classification_report(y_pred,y_test))

              precision    recall  f1-score   support

           0       0.56      0.55      0.56      1025
           1       0.55      0.52      0.54      1064
           2       0.33      0.38      0.35       880
           3       0.28      0.34      0.30       825
           4       0.39      0.39      0.39      1006
           5       0.39      0.43      0.41       907
           6       0.56      0.46      0.51      1216
           7       0.47      0.52      0.50       894
           8       0.61      0.58      0.60      1050
           9       0.52      0.46      0.49      1133

    accuracy                           0.47     10000
   macro avg       0.47      0.46      0.46     10000
weighted avg       0.48      0.47      0.47     10000



Thus it can be concluded that the Random Forest Classifier shows only 47% accuracy on the test set.